exercises/binary-search-tree/Example.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+public class BinarySearchTree<T>
+    where T : IComparable
+{
+    private BinarySearchTree(T value)
```

Comment:
While I understand some of the reasonings behind using static methods to create objects, I don't feel that it adds much value here.